assertj-core/src/test/java/org/assertj/core/api/DefaultAssertionErrorCollector_addAfterAssertionErrorCollected_Test.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2024 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+class DefaultAssertionErrorCollector_addAfterAssertionErrorCollected_Test {
+  private List<String> errorMessages;
+  private List<Throwable> throwables;
+  private SoftAssertions softly;
+
+  @BeforeEach
+  void given() {
+    errorMessages = new ArrayList<>();
+    throwables = new ArrayList<>();
+    softly = new SoftAssertions();
+  }
+
+  @Test
+  void should_perform_both_specified_actions_on_each_assertion_error() {
+    // WHEN
+    softly.addAfterAssertionErrorCollected(err -> errorMessages.add(err.getMessage()));
+    softly.addAfterAssertionErrorCollected(throwables::add);
+    softly.assertThat(1)
+          .isEqualTo(1_000)
+          .isBetween(7, 15)
+          .isEven();
+    // THEN
+    then(errorMessages).hasSameSizeAs(throwables)
+                       .hasSize(3);
+  }
+
+  @Test
+  void should_execute_both_specified_actions_on_each_manually_added_error() {
+    // WHEN
+    softly.addAfterAssertionErrorCollected(err -> errorMessages.add(err.getMessage()));
+    softly.addAfterAssertionErrorCollected(throwables::add);
+    softly.collectAssertionError(new AssertionError("hello"));
+    softly.collectAssertionError(new AssertionError("world"));
+    // THEN
+    then(errorMessages).hasSameSizeAs(throwables)
+                       .hasSize(2);
+  }
+
+  @Test
+  void should_register_the_same_callback_several_times() {
+    // GIVEN
+    AfterAssertionErrorCollected callback = throwables::add;
+    // WHEN
+    for (int i = 0; i < 10; i++) {
+      softly.addAfterAssertionErrorCollected(callback);
+    }
+    softly.collectAssertionError(new AssertionError("hello"));
+    // THEN
+    then(throwables).hasSize(10);
+  }
```

Comment:
add a unit test to check that setAfterAssertionErrorCollected replace all registered callback with the one specified

Reply:
Done. But I would like to clarify whether this test should be moved from this class 
`DefaultAssertionErrorCollector_addAfterAssertionErrorCollected_Test`
to this new one
`DefaultAssertionErrorCollector_setAfterAssertionErrorCollected_Test`
?
Or even moved to `org/example/test/DefaultAssertionErrorCollector_Test.java`
Or let it be in the current class, but rename the class to
`DefaultAssertionErrorCollector_afterAssertionErrorCollected_methods_Test`
or
`DefaultAssertionErrorCollector_callbacks_Test`
or something?